src/main/java/com/adventofcode/day16/Day16.java

```diff
@@ -0,0 +1,69 @@
+package com.adventofcode.day16;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Day16 {
+    private final List<Aunt> aunts;
+    private final Map<String, Integer> tickerTape;
+    private final static Pattern AUNT_PATTERN = Pattern.compile("Sue (\\d+): (\\w+): (\\d+), (\\w+): (\\d+), (\\w+): (\\d+)");
+
+    public Day16(List<String> input, Map<String, Integer> tickerTape) {
+        aunts = input.stream().map(this::parse).toList();
+        this.tickerTape = tickerTape;
+    }
+
+    public int partI() {
+        return findCorrectAunt(1).index();
+    }
+
+    public int partII() {
+        return findCorrectAunt(2).index();
+    }
+
+    private Aunt findCorrectAunt(int part) {
+        return aunts.stream()
+                .filter(
+                        aunt -> aunt.gifts().entrySet().stream()
+                                .allMatch(
+                                        entry -> {
+                                            if (part == 1) {
```

Comment:
I know why it's done this way, but two separated `findCorrectAunt` methods would be more readable